sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -399,6 +418,43 @@ def test_global_fanout(self):
           | beam.CombineGlobally(combine.MeanCombineFn()).with_fanout(11))
       assert_that(result, equal_to([49.5]))
 
+  def test_combining_with_accumulation_mode(self):
+    # PCollection will contain elements from 1 to 5.
+    elements = [i for i in range(1, 6)]
+
+    ts = TestStream().advance_watermark_to(0)
+    for i in elements:
+      ts.add_elements([i])
+    ts.advance_watermark_to_infinity()
+
+    options = PipelineOptions()
+    options.view_as(StandardOptions).streaming = True
+    with TestPipeline(options=options) as p:
+      result = (p
+                | ts
+                | beam.WindowInto(
+                    GlobalWindows(),
+                    accumulation_mode=trigger.AccumulationMode.ACCUMULATING,
+                    trigger=AfterWatermark(early=AfterAll(AfterCount(1)))
+                    )
+                | beam.CombineGlobally(sum).without_defaults().with_fanout(2)
+                | beam.ParDo(self.record_dofn()))
+
+    # The trigger should fire repeatedly for each newly added element,
+    # and at least once for advancing the watermark to infinity.
+    # The firings should accumulate the output.
+    # First firing: 1 = 1
+    # Second firing: 3 = 1 + 2
+    # Third firing: 6 = 1 + 2 + 3
+    # Fourth firing: 10 = 1 + 2 + 3 + 4
+    # Fifth firing: 15 = 1 + 2 + 3 + 4 + 5
+    # Next firings: 15 = 15 + 0  (advancing the watermark to infinity)
+    # The exact number of firings may vary,
```

Comment:
This should be deterministic. That's the point of TestStream. When does the last firing not happen?

Reply:
The firings were [1, 3, 6, 10, 15, 15] last week before I syn to the master.
And it became [1, 3, 6, 10, 15, 15, 15] after I fetch and rebase last Sunday.

Reply:
This was due to a bug fix for firing for discarding windows. The latter is correct.

Reply:
Please change this to expect the one correct answer.

Reply:
No need to record firings anymore. See comments above.